panel/widgets/chat.py

```diff
@@ -0,0 +1,1085 @@
+import asyncio
+import datetime
+
+from contextlib import ExitStack
+from dataclasses import dataclass
+from inspect import isawaitable
+from io import BytesIO
+from tempfile import NamedTemporaryFile
+from typing import (
+    Any, BinaryIO, ClassVar, Dict, List, Optional, Type, Union,
+)
+
+import param
+
+from ..io.resources import CDN_DIST
+from ..layout import (
+    Column, ListPanel, Row, Tabs,
+)
+from ..layout.card import Card
+from ..pane.base import panel as _panel
+from ..pane.image import PDF, Image, ImageBase
+from ..pane.markup import HTML, DataFrame, Markdown
+from ..pane.media import Audio, Video
+from ..reactive import ReactiveHTML
+from ..viewable import Viewable
+from .base import CompositeWidget, Widget
+from .button import Button
+from .indicators import LoadingSpinner
+from .input import FileInput, TextInput
+
+
+@dataclass
+class _FileInputMessage:
```

Comment:
Why do we not use Parameterized models here?

When would we generally recommend using dataclass or pydantic models or Parameterized models?

Reply:
dataclass seems most lightweight to me and it's just storing attributes here; it could be a param model here.

I don't think we would ever use pydantic models internally though.

Reply:
Ok